ecosystem/node-checker/README.md

```diff
@@ -4,42 +4,135 @@ The Aptos Node Health Checker (NHC) is the reference implementation of a node he
+
+## Running NHC: Source
+First, check out the source:
 ```
 git clone git@github.com:aptos-labs/aptos-core.git
-cd aptos-core/ecosystem/node-checker
+cd aptos-core
+```
+
+Depending on your setup, you may want to check out a particular branch, to ensure NHC is compatible with your node, e.g. `git checkout --track devnet`.
+
+From here, assuming you have followed the above configuration guide, you can run NHC:
+```
+cargo run --release -- server run --baseline-node-config-paths /etc/nhc/ait2_validator.yaml /etc/nhc/devnet_fullnode.yaml
+```
+
+## Running NHC: Terraform / Helm
+Down the line we will have easier pre-packaged configs in which you only need to specify key pieces of the configuration. Coming soon!
+
+## Running NHC as a sidecar
+When you run NHC as a sidecar, you preconfigure a node that NHC should use as the node under investigation by default:
+```
+--target-node-url http://localhost
+```
+
+Running NHC as a sidecar can be handy when you want to close the API / metrics ports on your machine to the public internet, but would still like to run NHC to validate the setup of your node.
+
+If you want, you can even restrict NHC to test only that node:
 ```
+--allow-preconfigured-test-node-only
+```
+With this flag, the `/check_node` endpoint will always return 400s, you must instead use `/check_preconfigured_node`.
 
-Next, generate a baseline configuration. In this example we generate a configuration for a FullNode on the devnet, running on a machine in the local network:
+Once you have configured your NHC instance in sidecar mode, you can send requests that omit the target node address.
 ```
-cargo run -- configuration create --configuration-name "My Server Devnet Fullnode" --url http://192.168.86.2 --evaluators state_sync -o /tmp/my_server_devnet_fullnode.yaml
+curl 'http://nhc.aptoslabs.com/check_preconfigured_node?baseline_configuration_name=devnet_fullnode'
 ```
 
-Finally, run NHC:
+There are more options than these, e.g. around which ports to use. Pass `-h` to see more options.
+
+---
+
+## Generating the OpenAPI specs
+To generate the OpenAPI specs, run the following commands:
 ```
-cargo run -- server run --baseline-node-config-paths /tmp/my_server_devnet_fullnode.yaml
+cargo run -- server generate-openapi -f yaml > openapi.yaml
+cargo run -- server generate-openapi -f json > openapi.json
 ```
-Where `--baseline-node-url` is the node that will be used as the baseline against which NHC will compare the test node (the node under investigation).
 
-Instructions for running the tool using Docker / Terraform are coming soon.
+You can also hit the `spec_yaml` and `spec_json` endpoints of the running service.
 
 ## Developing
 To develop this app, you should first run two nodes of the same type. See [this wiki](https://aptos.dev/tutorials/full-node/run-a-fullnode) for guidance on how to do this. You may also target a known existing FullNode with its metrics port open.
```

Comment:
"this app", what app this is referring to? This section did a sudden duck shuffle and lost me. Also, the link to running a FullNode should be fixed to: `https://aptos.dev/nodes/full-node/fullnode-for-devnet`.